tests/dedicatedevhtests/l7_evh_crd_test.go

```diff
@@ -0,0 +1,440 @@
+	g.Expect(*nodes[0].OauthVsConfig.RedirectURI).To(gomega.Equal("https://auth.com/oauth/redirect"))
+	g.Expect(*nodes[0].OauthVsConfig.PostLogoutRedirectURI).To(gomega.Equal("https://auth.com/oauth/post-logout-redirect"))
+	g.Expect(nodes[0].OauthVsConfig.OauthSettings).To(gomega.HaveLen(1))
+	g.Expect(*nodes[0].OauthVsConfig.OauthSettings[0].AppSettings.ClientID).To(gomega.Equal("my-client-id"))
+	g.Expect(*nodes[0].OauthVsConfig.OauthSettings[0].AppSettings.ClientSecret).To(gomega.Equal("my-client-secret"))
+	g.Expect(nodes[0].OauthVsConfig.OauthSettings[0].AppSettings.Scopes).To(gomega.HaveLen(1))
+	g.Expect(nodes[0].OauthVsConfig.OauthSettings[0].AppSettings.Scopes[0]).To(gomega.Equal("scope-1"))
+	g.Expect(*nodes[0].OauthVsConfig.OauthSettings[0].AppSettings.OidcConfig.OidcEnable).To(gomega.Equal(true))
+	g.Expect(*nodes[0].OauthVsConfig.OauthSettings[0].AppSettings.OidcConfig.Profile).To(gomega.Equal(true))
+	g.Expect(*nodes[0].OauthVsConfig.OauthSettings[0].AppSettings.OidcConfig.Userinfo).To(gomega.Equal(true))
+	g.Expect(*nodes[0].OauthVsConfig.OauthSettings[0].AuthProfileRef).To(gomega.ContainSubstring("thisisaviref-authprofileoauth"))
+	g.Expect(*nodes[0].OauthVsConfig.OauthSettings[0].ResourceServer.AccessType).To(gomega.Equal("ACCESS_TOKEN_TYPE_OPAQUE"))
+	g.Expect(*nodes[0].OauthVsConfig.OauthSettings[0].ResourceServer.IntrospectionDataTimeout).To(gomega.Equal(int32(60)))
+	g.Expect(*nodes[0].OauthVsConfig.OauthSettings[0].ResourceServer.OpaqueTokenParams.ServerID).To(gomega.Equal("my-server-id"))
+	g.Expect(*nodes[0].OauthVsConfig.OauthSettings[0].ResourceServer.OpaqueTokenParams.ServerSecret).To(gomega.Equal("my-server-secret"))
+	g.Expect(nodes[0].OauthVsConfig.OauthSettings[0].ResourceServer.JwtParams).To(gomega.BeNil())
+	g.Expect(nodes[0].SamlSpConfig).To(gomega.BeNil())
+
+	//Update with Oidc parameters as false
+	srUpdate := integrationtest.FakeSSORule{
+		Name:      srname,
+		Namespace: "default",
+		Fqdn:      "foo.com",
+		SSOType:   "OAuth",
+	}.SSORule()
+	srUpdate.ResourceVersion = "2"
+	oidcEnable, profile, userinfo := false, false, false
+	srUpdate.Spec.OauthVsConfig.OauthSettings[0].AppSettings.OidcConfig = &v1alpha2.OIDCConfig{
+		OidcEnable: &oidcEnable,
+		Profile:    &profile,
+		Userinfo:   &userinfo,
+	}
+	_, err = v1alpha2CRDClient.AkoV1alpha2().SSORules("default").Update(context.TODO(), srUpdate, metav1.UpdateOptions{})
+	if err != nil {
+		t.Fatalf("error in updating SSORule: %v", err)
+	}
+	g.Eventually(func() string {
+		ssoRule, _ := v1alpha2CRDClient.AkoV1alpha2().SSORules("default").Get(context.TODO(), srname, metav1.GetOptions{})
+		return ssoRule.Status.Status
+	}, 10*time.Second).Should(gomega.Equal("Accepted"))
+
+	g.Eventually(func() int {
+		_, aviModel := objects.SharedAviGraphLister().Get(modelName)
+		nodes, ok := aviModel.(*avinodes.AviObjectGraph)
+		if !ok {
+			return 0
+		}
+		return len(nodes.GetAviEvhVS())
+	}, 20*time.Second).Should(gomega.Equal(1))
```

Comment:
This code is of no use since the model will be already present.

Reply:
removed